Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -995,6 +1025,67 @@ open class ComposerVC: _ViewController,
         NotificationCenter.default.post(name: .createNewPoll, object: nil, userInfo: userInfo)
     }
 
+    @objc func sendWeatherMessage(_ notification: Notification ) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) { [weak self] in
+            guard let `self` = self else { return }
+            self.channelController?
+                .createNewMessage(
+                    text: "/weather",
+                    extraData: ["weather": .dictionary(self.getWeatherRequest(notification))],
+                    completion: nil
+                )
+        }
+    }
+
+    @objc func editWeatherMessage(_ notification: Notification ) {
+        let messageId = notification.userInfo?["messageId"] as? String ?? ""
+        guard let cid = channelController?.cid else { return }
+        let action = AttachmentAction(name: "action", value: "edit", style: .default, type: .button, text: "Edit")
+        ChatClient.shared.messageController(cid: cid, messageId: messageId)
+            .dispatchEphemeralMessageAction(action, getWeatherRequest(notification), key: "weather")
+    }
+
+    func getWeatherRequest(_ notification: Notification) -> [String: RawJSON] {
+        var weatherData = [String: RawJSON]()
+        // Get data from notification
+        let temp = notification.userInfo?["tempreature"] as? String ?? ""
+        let location = notification.userInfo?["location"] as? String ?? ""
+        let iconCode = notification.userInfo?["iconCode"] as? String ?? ""
+        let displayMessage = notification.userInfo?["displayMessage"] as? String ?? ""
+        let description = notification.userInfo?["description"] as? String ?? ""
+        // Set data
+        weatherData["currentWeather"] = .string(temp)
+        weatherData["currentLocation"] = .string(location)
+        weatherData["iconCode"] = .string(iconCode)
+        weatherData["displayMessage"] = .string(displayMessage)
+        weatherData["description"] = .string(description)
+        return weatherData
+    }
+
+    private func showWeather() {
+        isWeatherButtonTapped = false
+        guard let currentLocation = locationManager.location else { return }
+        LocationHelper.shared.getLocationInfo(currentLocation: currentLocation) { [weak self] placemark, error in
+            guard let self = self else { return }
+            var weatherData: [String: Any] = [
+                "latitude": currentLocation.coordinate.latitude,
+                "longitude": currentLocation.coordinate.longitude,
+                "locationName": placemark?.first?.locality]
+            NotificationCenter.default.post(name: .fetchWeather, object: nil, userInfo: weatherData)
+        }
+    }
+
+    private func showSettingsAlert() {
+        let alert = UIAlertController(title: "Permission missing", message: "Please allow location permission from settings to access this feature.", preferredStyle: .alert)
```

Comment:
@mdhanif-simformsolutions  Location Manager class already has these Alert. Please use it and remove duplications